### DISCUSSION TITLE
如何快速将自定义的插件构建到使用呢？

### DISCUSSION THREAD

**Discussion by nimohunter** [apache/apisix#6376] [Ideas]
* 解决办法1 ： 利用Oteam 的Tapisix的docker image，将这个镜像作为基础镜像，将自定义的plugin复制进来，生成自定义的镜像。
* 解决办法2 ： 修改Oteam 的Tapisix的dockerfile，override dockerfile中的cmd， 将wget plugin放入启动项来。（但是这种方法过于繁复，不便操作）
* 解决方法3 ： 修改Oteam 的Tapisix的代码，在生成config文件的同时，wget 自定义的plugin。（也许可行，不过镜像运行中再wget东西是否合乎docker的原则，值得考虑）
* 解决方法4 ： 直接fork Tapisix的代码， 本地进行修改代码进行构建镜像。

关于解决方法2和解决方法3，不知道各位大佬怎么看？

**Comment by tzssangglass** [comment]
I recommend using  `extra_lua_path `

see: https://github.com/apache/apisix/blob/master/docs/en/latest/plugin-develop.md#where-to-put-your-plugins
